File: src/test/java/com/github/ccguyka/showupdates/ArtifactBuilder.java
package com.github.ccguyka.showupdates;

import org.apache.maven.artifact.Artifact;
import org.apache.maven.artifact.DefaultArtifact;

public class ArtifactBuilder {

    private String version;
    private String groupId;
    private String artifactId;

    private ArtifactBuilder() {
        // prevent instantiation
    }

    public static ArtifactBuilder aParent() {
        return new ArtifactBuilder().groupId("parent-groupId").artifactId("parent-artifactId");
    }

    public static ArtifactBuilder aDependency() {
        return new ArtifactBuilder().groupId("dep-groupId").artifactId("dep-artifactId");
    }

    public static ArtifactBuilder aManagedDependency() {
        return new ArtifactBuilder().groupId("dep-mgnt-groupId").artifactId("dep-mgnt-artifactId");
    }

    public static ArtifactBuilder aPlugin() {
        return new ArtifactBuilder().groupId("plugin-groupId").artifactId("plugin-artifactId");
    }

    public ArtifactBuilder version(String version) {
        this.version = version;

        return this;
    }

    public ArtifactBuilder groupId(String groupId) {
        this.groupId = groupId;

        return this;
    }

    public ArtifactBuilder artifactId(String artifactId) {
        this.artifactId  = artifactId;

        return this;
    }

    public Artifact build() {
        return new DefaultArtifact(groupId, artifactId, version, "compile", "type", "classifier", null);
    }
}

File: src/test/kotlin/com/github/ccguyka/showupdates/ArtifactBuilder.kt
package com.github.ccguyka.showupdates

import org.apache.maven.artifact.Artifact
import org.apache.maven.artifact.DefaultArtifact

class ArtifactBuilder private constructor() {

    private var version: String? = null
    private var groupId: String? = null
    private var artifactId: String? = null

    fun version(version: String?): ArtifactBuilder {
        this.version = version
        return this
    }

    fun groupId(groupId: String?): ArtifactBuilder {
        this.groupId = groupId
        return this
    }

    fun artifactId(artifactId: String?): ArtifactBuilder {
        this.artifactId = artifactId
        return this
    }

    fun build(): Artifact {
        return DefaultArtifact(groupId, artifactId, version, "compile", "type", "classifier", null)
    }

    companion object {
        @JvmStatic
        fun aParent(): ArtifactBuilder {
            return ArtifactBuilder().groupId("parent-groupId").artifactId("parent-artifactId")
        }

        @JvmStatic
        fun aDependency(): ArtifactBuilder {
            return ArtifactBuilder().groupId("dep-groupId").artifactId("dep-artifactId")
        }

        @JvmStatic
        fun aManagedDependency(): ArtifactBuilder {
            return ArtifactBuilder().groupId("dep-mgnt-groupId").artifactId("dep-mgnt-artifactId")
        }

        @JvmStatic
        fun aPlugin(): ArtifactBuilder {
            return ArtifactBuilder().groupId("plugin-groupId").artifactId("plugin-artifactId")
        }
    }
}
